### PR TITLE
Add an ID generator

### PR DESCRIPTION
**Pull request by TrinityDevelopers**

### CODE REVIEW

**Comment by byteandahalf** [comment]
You can't check if the Block is NULL because MCPE fills in NULL spaces with update blocks AFAIK :(

**Comment by byteandahalf** [comment]
Does this work for you?

**Comment by TrinityDevelopers** [comment]
I haven't tested it yet, but it does work with items... does that make much of a difference?

**Comment by byteandahalf** [comment]
Yes it does, because all empty Blocks in the game are filled in with update blocks. I think I can fix this later.

**Comment by ghost** [comment]
You could check if the block name is empty then the id usable...

**Comment by TrinityDevelopers** [comment]
That sounds like a good idea.

**Comment by byteandahalf** [comment]
Actually, just use Block::mOwnesBlocks, that is a vector that only holds existing blocks

**Comment by ghost** [comment]
What would happen if no id is found? Right now I think it will crash

**Comment by TrinityDevelopers** [comment]
How would it crash?

**Comment by rekmolly** [comment]
TrinityDevelopers, You may suggest how can I find such things as: char filler1[56];, int type;, std::string filepath; and other

**Comment by ghost** [comment]
You are running through all the id, right? from 0 - INT_MAX
If the client have multiple mods installed and those mods use a lot of block id. In the loop the id will get higher than 256 which is the size of the Block:m Blocks array. When you try to add the block to the array with a id higher than 256 the game will crash

**Comment by TrinityDevelopers** [comment]
ok, I will make a fix for that

**Comment by TrinityDevelopers** [comment]
And @enoter do you have a twitter?  I can explain it for you on dm.

**Comment by TrinityDevelopers** [comment]
In addition, is there a reason why the block array cannot be extended?
If we make it 32,767, then the id gen can pass over all ids from 256 - 4096 (for items) and allow all ids for blocks from 4097 - 32,767.  What prevents this?

**Comment by ghost** [comment]
The things with block is that the block id is stored in one byte so the id can't exceed 255. Unless you want to rewrite the hole level system there is no way we can "extend" the array

File: jni/com/byteandahalf/ports/gc/core/blocks/GCBlocks.h [hunk 1]
```diff
@@ -10,6 +10,8 @@ class GCBlocks {
 	static void registerBlocks();
 	static void registerBlock(Block*);
 	
+	static void getNewRandomID();
```
Comment (InusualZ) [inline jni/com/byteandahalf/ports/gc/core/blocks/GCBlocks.h:13]: Return type: int

**Comment by rekmolly** [comment]
@TrinityDevelopers Yea, https://twitter.com/block_zone

**Pull request closed without merging**